### PR TITLE
Revert "Revert "Upgrade junit2jira (#5672)""

### PR DESCRIPTION
**Pull request by janisz**
Reverts stackrox/stackrox#5742

### CODE REVIEW

File: scripts/ci/lib.sh [hunk 1]
```diff
@@ -1223,9 +1223,17 @@ store_test_results() {
     if ! is_in_PR_context; then
     {
         info "Creating JIRA task for failures found in $from"
```
Comment (gavin-stackrox) [inline scripts/ci/lib.sh:1225]: what about wrapping the {} with a set +u, set -u? so we avoid any undefined causing a script exit?

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

File: scripts/ci/lib.sh (hunk 1 above)
Comment (porridge) [reply scripts/ci/lib.sh:1225]: Or even better, just explicitly provide defaults in each variable invocation. We might be able to come up with better values than empty strings at least in some cases.

**Review by porridge — CHANGES_REQUESTED** [review]

File: scripts/ci/lib.sh (hunk 1 above)
Comment (janisz) [reply scripts/ci/lib.sh:1226]: I'm not sure if explicit defaults are better as they are harder to read and empty string is good enough in our case

**Review by gavin-stackrox — APPROVED** [review]
LGTM

**Comment by janisz** [comment]
@porridge PTAL

File: scripts/ci/lib.sh (hunk 1 above)
Comment (porridge) [reply scripts/ci/lib.sh:1226]: I don't know whether I agree with the "good enough" part, but meh..

**Review by porridge — APPROVED** [review]

**Pull request merged**